models/topics.models.js

```diff
@@ -1,7 +1,16 @@
 const db = require("../db/connection");
 
-exports.selectTopics = () => {
-  return db.query(`SELECT * FROM topics`).then(({ rows }) => {
+exports.selectTopics = (topic) => {
+  const queryVals = [];
+  let sqlString = `SELECT * FROM topics`;
+  if (topic) {
+    sqlString += ` WHERE slug = $1`;
+    queryVals.push(topic);
+  }
+  return db.query(sqlString, queryVals).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not Found" });
```

Comment:
Since you've made this custom error, and you know exactly what causes it (a topic is not found) you can afford to be a bit more specific with the msg. This helps on the frontend so you can tell the difference between 404 messages, and have a better idea of what exactly causes each one instead of throwing them all into the same generic bucket